cmd/agent/daemon/app/app.go

```diff
@@ -146,41 +146,42 @@ func (a *App) Run(ctx context.Context) error {
 		if err := a.syncRemoteConfig(ctx, castaiClient); err != nil {
 			return fmt.Errorf("sync remote config: %w", err)
 		}
-		if a.cfg.SendLogsLevel != "" && a.cfg.Castai.Valid() {
+		if cfg.SendLogsLevel != "" {
 			castaiLogsExporter := castai.NewLogsExporter(castaiClient)
 			go castaiLogsExporter.Run(ctx) //nolint:errcheck
 
-			if a.cfg.PromMetricsExportEnabled {
+			if cfg.PromMetricsExportEnabled {
 				castaiMetricsExporter := castai.NewPromMetricsExporter(log, castaiLogsExporter, prometheus.DefaultGatherer, castai.PromMetricsExporterConfig{
 					PodName:        podName,
-					ExportInterval: a.cfg.PromMetricsExportInterval,
+					ExportInterval: cfg.PromMetricsExportInterval,
 				})
 				go castaiMetricsExporter.Run(ctx) //nolint:errcheck
 			}
 
 			logCfg.Export = logging.ExportConfig{
 				ExportFunc: castaiLogsExporter.ExportFunc(),
-				MinLevel:   logging.MustParseLevel(a.cfg.SendLogsLevel),
+				MinLevel:   logging.MustParseLevel(cfg.SendLogsLevel),
 			}
-			log = logging.New(logCfg)
 		}
+		log = logging.New(logCfg)
```

Comment:
This is the real fix, initialize the logger always before doing `state.NewExporters(log)`